core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -282,20 +283,77 @@ private static boolean isCreate(UpdateTableRequest request) {
     return isCreate;
   }
 
-  private static TableMetadata create(TableOperations ops, TableMetadata start, UpdateTableRequest request) {
+  private static TableMetadata commitUpdate(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    Table table = catalog.loadTable(ident);
+    if (table instanceof BaseTable) {
+      TableOperations ops = ((BaseTable) table).operations();
+      return commit(ops, request.requirements(), request.updates());
+    } else {
+      throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+    }
+  }
+
+  private static TableMetadata commitCreate(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    // this is a hacky way to get TableOperations for an uncommitted table
+    Transaction transaction = catalog.buildTable(ident, EMPTY_SCHEMA).createOrReplaceTransaction();
+    if (!(transaction instanceof BaseTransaction)) {
+      throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTransaction");
```

Comment:
Looks like this could be cleaner if it were rewritten to use `Preconditions.checkState` instead.

Reply:
I made this change.